splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/TableScanOperation.java

```diff
@@ -175,10 +190,34 @@ public TableScanOperation(long conglomId,
         this.indexColItem=indexColItem;
         this.indexName=indexName;
         init();
+        if(pastTxFunctor != null) {
+            this.pastTx = MapToTxId((DataValueDescriptor)pastTxFunctor.invoke(activation));
+            if(pastTx == -1) {
+                pastTx = 0; // force going back to oldest transaction instead of ignoring it.
```

Comment:
Txn id 0 is used for the ROOT_TRANSACTION, which is a special transaction parent of all user level transactions. I don't think this is going to be problematic but I'd rather use a different transaction id other than 0, in case we think at some point this is actually the root transaction.

Reply:
Done. I created a constant: `public static final long OLDEST_TIME_TRAVEL_TX = TRASANCTION_INCREMENT;` and I am using it here instead of simply using zero.